lib/MusicBrainz/Server/Data/Relationship.pm

```diff
@@ -230,6 +230,55 @@ sub _load_related_info {
     $self->load_entities(@rels);
 }
 
+sub load_credits
+{
+    my ($self, $type, @ids) = @_;
+    my @types = map { [ sort($type, $_) ] } @RELATABLE_ENTITIES;
+    my @credits;
+    foreach my $t (@types) {
+        my $type0 = $t->[0];
+        my $type1 = $t->[1];
+        my $query;
+        my @params;
+
+        my $type0_query = <<~"SQL";
+            SELECT entity0_credit
+              FROM l_${type0}_${type1}
+             WHERE entity0 = any(?)
+               AND entity0_credit != ''
+            SQL
+
+        my $type1_query = <<~"SQL";
+            SELECT entity1_credit
+              FROM l_${type0}_${type1}
+             WHERE entity1 = any(?)
+               AND entity1_credit != ''
+            SQL
+
+        if ($type eq $type0 && $type eq $type1) {
+            $query = <<~"SQL";
+                (
+                    $type0_query
+                ) UNION (
+                    $type1_query                    
+                )
+                SQL
+            push @params, @ids, @ids;
+        } elsif ($type eq $type0) {
+            $query = $type0_query;
+            push @params, @ids;
+        } elsif ($type eq $type1) {
+            $query = $type1_query;
+            push @params, @ids;
+        }
+
+        my $results = $self->sql->select_single_column_array($query, @params);
+        push @credits, @$results;
+    }
+    my @sorted_credits = uniq sort {$a cmp $b} @credits;
```

Comment:
`UNION` already eliminates duplicates.
```suggestion
    my @sorted_credits = sort { $a cmp $b } @credits;
```

Reply:
But there might be many relationships with the same credit, so we still need to take care of those. I guess the ones without the UNION should do SELECT DISTINCT in that case?

Reply:
Oh, no, actually that won't work, because the SQL queries are *per entity*. `uniq` here ensures that there are no duplicates also *between different relationship tables*. Not sure if doing DISTINCT *as well* is good tho, for `uniq` to have to run over less rows?

Reply:
Oh, for some reason I was thinking there was only one query. Can you merge them into one? They all only select a single column of the same type, so should all `UNION` fine. (You don't need any `DISTINCT` or `uniq` in that case.)